core/trino-main/src/main/java/io/trino/sql/planner/planprinter/HashCodeAnonymizer.java

```diff
@@ -0,0 +1,350 @@
+                target.getHandle().getCatalogName().getCatalogName(),
+                target.getSchemaTableName().getSchemaName(),
+                target.getSchemaTableName().getTableName());
+        return anonymize(qualifiedObjectName);
+    }
+
+    private String anonymize(InsertTarget target)
+    {
+        QualifiedObjectName qualifiedObjectName = new QualifiedObjectName(
+                target.getHandle().getCatalogName().getCatalogName(),
+                target.getSchemaTableName().getSchemaName(),
+                target.getSchemaTableName().getTableName());
+        return anonymize(qualifiedObjectName);
+    }
+
+    private String anonymize(RefreshMaterializedViewTarget target)
+    {
+        QualifiedObjectName qualifiedObjectName = new QualifiedObjectName(
+                target.getInsertHandle().getCatalogName().getCatalogName(),
+                target.getSchemaTableName().getSchemaName(),
+                target.getSchemaTableName().getTableName());
+        return anonymize(qualifiedObjectName);
+    }
+
+    private String anonymize(DeleteTarget target)
+    {
+        QualifiedObjectName qualifiedObjectName = new QualifiedObjectName(
+                target.getHandle().map(TableHandle::getCatalogName).map(CatalogName::getCatalogName).orElse(""),
+                target.getSchemaTableName().getSchemaName(),
+                target.getSchemaTableName().getTableName());
+        return anonymize(qualifiedObjectName);
+    }
+
+    private String anonymize(UpdateTarget target)
+    {
+        QualifiedObjectName qualifiedObjectName = new QualifiedObjectName(
+                target.getHandle().map(TableHandle::getCatalogName).map(CatalogName::getCatalogName).orElse(""),
+                target.getSchemaTableName().getSchemaName(),
+                target.getSchemaTableName().getTableName());
+        return anonymize(qualifiedObjectName);
+    }
+
+    private String anonymize(TableExecuteTarget target)
+    {
+        QualifiedObjectName qualifiedObjectName = new QualifiedObjectName(
+                target.getExecuteHandle().getCatalogName().getCatalogName(),
+                target.getSchemaTableName().getSchemaName(),
+                target.getSchemaTableName().getTableName());
+        return anonymize(qualifiedObjectName);
+    }
+
+    /**
+     * Rounds n to the nearest order of magnitude with base 10.
+     */
+    private static double round10(double n)
+    {
+        return pow(10, round(log10(n)));
+    }
+
+    private static String formatMap(Map<String, String> map)
+    {
+        return map.entrySet().stream()
+                .map(entry -> entry.getKey() + " = " + entry.getValue())
+                .collect(joining(", ", "[", "]"));
+    }
+
+    private <T> String anonymize(T object, ObjectType objectType)
+    {
+        return objectType.name().toLowerCase(Locale.ENGLISH) + "_" + Integer.toUnsignedLong((environment + object.toString()).hashCode());
```

Comment:
I think you should use some strong hash here for security purpose.

I would use query id instead of `environment` for salt. It changes for every query

Reply:
Switching to query id for salt would mean that we cannot know whether exactly the same query is being repeated in an environment, we will only be able to know whether there is repetition in query shape. It'll also be harder to compare EXPLAINs from multiple runs of same query.

Reply:
I've updated the hashing method to `md5`

Reply:
We should use `Hashing.goodFastHash(64)` if we don't want to rely on Object#hashCode.
However, that would mean we don't benefit from caching of hash codes by String.
Please do some local JFR profiling by running TestHiveTpcdsCostBasedPlan with this functionality turned on to confirm that we're not adding any significant overheads from anonymisation.

Reply:
I've updated the method to `murmur3_128`